### ISSUE TITLE
False positive on MD052 when square brackets are present in the URL

### ISSUE THREAD

**Issue by ojob** [DavidAnson/markdownlint#697]
I have a file with following links definitions (note the square brackets in the URLs):

```markdown

- [Board Fonctionnel][functional-board] : suivi des fonctions à implémenter
- [Board Développement][dev-board] : suivi de la conception et du développement

[functional-board]: https://gitlab.xxxxxxxx/-/boards/4?label_name[]=type:business
[dev-board]: https://gitlab.xxxxxxx/-/boards/5?not[label_name][]=type:business&not[label_name][]=type:meta

```

I just upgraded to [Markdownlint-cli 0.33.0](), based on `markdownlint` 0.27.0, and now I get the following message against the lines above :

    README.md:17:89 MD052/reference-links-images Reference links and images should use a label that is defined [Missing link or image reference definition: "label_name"] [Context: "[label_name][]"]
    README.md:17:121 MD052/reference-links-images Reference links and images should use a label that is defined [Missing link or image reference definition: "label_name"] [Context: "[label_name][]"]

This is a false alert, as `[label_name][]` is no link anchor, but a part of the address.

So for the moment, I disabled the rule, but I think that `markdownlint` should not look for links inside a line defining a link.

**Comment by nschonni** [comment]
You're using `[functional-board]` as the anchor link, but `[dev-board]` in the text, so it is flagging a problem

**Comment by ojob** [comment]
Ah, sorry that's an issue with my copy-paste, hum! I updated the report.

**Comment by ojob** [comment]
Thanks! I look forward the update to reenable the rule.

**Comment by DavidAnson** [comment]
It's available in the latest library now; the CLI update will likely be in a few weeks.
